app/models/queries/work_packages/filter/duration_filter.rb

```diff
@@ -0,0 +1,42 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class Queries::WorkPackages::Filter::DurationFilter <
+  Queries::WorkPackages::Filter::WorkPackageFilter
+  def type
+    :integer
+  end
+
+  def where
+    if operator == Queries::Operators::None.to_sym.to_s
+      super + " OR #{WorkPackage.table_name}.duration=0"
+    else
+      super
+    end
```

Comment:
It is not really clear to me what to do with milestone typed work packages. Since they are not supposed to have a duration, as in they do not have the property, they probably should never be returned. This might be different for the `None` operator since then the notion would be 'Work packages without duration' which in my opinion includes milestones. 

Currently this is even harder since milestones still have a duration of 1 as of now.

I am open to suggestions here.

Reply:
DONE: b9994104d1143a7676ae74b47c6c286c7edc8245

It does not return the milestone type wps, when the filter is on, but it does return them, when the filter is "none".

Reply:
👍